### PR TITLE
Feature/ansible lint circleci

### LINKED ISSUE #65: Feature/ansible-lint
---
### Feature

- as a developer, I need to make sure that the ansible best practice are respected on playbook

---
### Motivation

- [ ] Ansible-lint checks playbooks for practices and behaviour that could potentially be improved

### PR DESCRIPTION
**Pull request by guiadco**
### Current behaviour

Actually i couldn't know if i respect ansible best practice, so i could do anything i want.
There's no rules

---
### Expected behaviour

I want rules to produce a better code for the community

---
### Modifications

> What are the changes involved to match with the expected behaviour ?

-  [ ] Refactoring allspark architecture folder
-  [ ] Split all test task with a workflow on circleci
-  [x] An ansible lint config adding to give the default best practices
-  [x] Only Ansible playbook and role are checked
-  [ ] The documentation has been updated

---
### Related issues

> What issue are affected by or affecting this pull request.

- resolve #65 

---
### Status

- [x] Implementation
- [x] Test coverage
- [ ] Documentation

### CODE REVIEW

**Comment by guiadco** [comment]
```
#!/bin/bash -eo pipefail
PATH=$PATH:/root/.local/bin ansible-lint install.yml -c .circleci/.ansible-lint

Examining install.yml of type playbook
Examining /root/repo/roles/system/tasks/main.yml of type tasks
Examining /root/repo/roles/pip/tasks/main.yml of type tasks
Examining /root/repo/roles/download/tasks/main.yml of type tasks
Examining /root/repo/roles/infra/tasks/main.yml of type tasks
Examining /root/repo/roles/monitoring/tasks/main.yml of type tasks
Examining /root/repo/roles/gitlab/tasks/main.yml of type tasks
Examining /root/repo/roles/jenkins/tasks/main.yml of type tasks
Examining /root/repo/roles/sonarqube/tasks/main.yml of type tasks
Examining /root/repo/roles/system/tasks/rhel7.yml of type tasks
/root/repo/roles/pip/tasks/main.yml:13: [EANSIBLE0002] Trailing whitespace
/root/repo/roles/system/tasks/rhel7.yml:19: [EANSIBLE0010] Package installs should not use latest
/root/repo/roles/system/tasks/rhel7.yml:25: [EANSIBLE0010] Package installs should not use latest
/root/repo/roles/system/tasks/rhel7.yml:31: [EANSIBLE0010] Package installs should not use latest
/root/repo/roles/system/tasks/rhel7.yml:55: [EANSIBLE0013] Use shell only when shell functionality is required
/root/repo/roles/system/tasks/rhel7.yml:67: [EANSIBLE0013] Use shell only when shell functionality is required
```

File: .circleci/.ansible-lint [hunk 1]
```diff
@@ -0,0 +1,10 @@
+parseable: true
+quiet: false
+use_default_rules: true
+verbosity: 1
+exclude_paths:
```
Comment (Xide) [inline .circleci/.ansible-lint]: Change from the old pull request was not applied here as described [Here](https://github.com/actiniumio/allspark/pull/66/files#r209425288).

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - checkout
+      - add_ssh_keys:
+          fingerprints:
+            - "59:35:d6:58:31:87:ac:e6:f2:87:f3:0e:9e:8d:c2:e1"
+      - run: pip install --user -r ./.circleci/requirements.txt
+      - run: PATH=$PATH:/root/.local/bin ansible-lint install.yml -c .circleci/.ansible-lint
+      - run: PATH=$PATH:/root/.local/bin ansible-lint setup.yml -c .circleci/.ansible-lint
+      - run: PATH=$PATH:/root/.local/bin ansible-lint release.yml -c .circleci/.ansible-lint
+      - run: PATH=$PATH:/root/.local/bin ansible-lint restore.yml -c .circleci/.ansible-lint
+      - run: PATH=$PATH:/root/.local/bin ansible-lint backup.yml -c .circleci/.ansible-lint
```
Comment (Xide) [inline .circleci/config.yml]: An issue with this implementation is that we have to remember to add a test line when we create a new playbook.
I think we either need to document this in the `CONTRIBUTING.md` or to put the playbooks in a separated folder and use a wildcard matching.

Comment (guiadco) [reply .circleci/config.yml]: I will see to refactor the architecture in another PR so i will edit the ```CONTRIBUTING.md```

Comment (guiadco) [reply .circleci/config.yml]: those playbook are the main core of allspark so normally you don't have to often change it.
the other are store on role/**/task/** so all the future implementation will be track automatically

**Pull request merged**